doc/riscv_trace.rst

```diff
@@ -0,0 +1,70 @@
+|                    |                     | 3: Report as asynchronous discontinuity.                    |
++--------------------+---------------------+-------------------------------------------------------------+
+| rv_trace_itype     |           3         | 0: Final instruction in block is none of other itype codes; |
+|                    |                     | 1: Exception. An exception that traps occurred              |
+|                    |                     | following the final retired instruction in the block;       |
+|                    |                     | 2: Interrupt. An interrupt that traps occurred              |
+|                    |                     | following the final retired instruction in the block;       |
+|                    |                     | 3: Exception or interrupt return;                           |
+|                    |                     | 4: Nontaken branch;                                         |
+|                    |                     | 5: Taken branch;                                            |
+|                    |                     | 6: Uninferable jump if itype_width_p is 3                   |
++--------------------+---------------------+-------------------------------------------------------------+
+| rv_trace_iaddr     |          32         | Contains the value of the address in the cases it needs     |
+|                    |                     | to be reported in. Namely:                                  |
+|                    |                     | -Jumps with target that can't be inferred from source code; |
+|                    |                     | -The instruction retired immediately after a jump with a    |
+|                    |                     | target that cannot be inferredfrom the source code (also    |
+|                    |                     | referred to as the target or destination of the jump);      |
+|                    |                     | -Taken and nontaken branches;                               |
+|                    |                     | -Last instruction retired before an exception or interrupt; |
+|                    |                     | -First instruction retired following an exception or intert;|
+|                    |                     | -The last instruction retired before a privilege change;    |
+|                    |                     | -The first instruction retired following a privilege change.|
++--------------------+---------------------+-------------------------------------------------------------+
+| rv_trace_cause     |          12         | ucause/scause/mcause CSR                                    |
++--------------------+---------------------+-------------------------------------------------------------+
+| rv_trace_tval      |          32         | Associated trap value, e.g. the faulting virtual address for|
```

Comment:
- Ibex doesn't implement S mode, so there's no stval.
- Ibex doesn't support virtual memory, so instead of virtual address, it's just address.